PhotoStream/app/src/main/java/com/keenanzucker/photostream/MainActivity.java

```diff
@@ -0,0 +1,72 @@
+package com.keenanzucker.photostream;
+
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.app.FragmentTransaction;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.Menu;
+import android.view.MenuItem;
+
+/**
+ * Main Activity for Photostream Application.
+ * Uses two Fragments, SearchFragment that searches for images and displays images one by one, offering choices
+ * to go the next image, save the image, or access the image library.
+ * The image library is contained in StreamFragment, which steams the saved images in a gridview. Images can be deleted with
+ * a long click. Contains an option to go back to the searchFragment as well.
+ *
+ * Main Activity contains transitions to each fragments
+ */
+public class MainActivity extends AppCompatActivity implements SearchFragment.onLibraryListener, StreamFragment.onBackButtonListener {
```

Comment:
I would keep your interfaces in separate files as oppose to within a class such as SearchFragment